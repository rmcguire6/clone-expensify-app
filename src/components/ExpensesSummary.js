import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';
export class ExpensesSummary extends React.Component {

  render() {
    return (
      <div>
        {this.props.expenses &&
          <p>Viewing {this.props.expenses.length}{this.props.expenses.length === 1 ? ` expense` : ` expenses`} totaling   {numeral(this.props.expensesTotal / 100).format('$0,0.00')}</p>}
      </div>
    )
  };
};
const mapStoreToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters),
    expensesTotal: selectExpensesTotal(selectExpenses(state.expenses, state.filters))
  };
}
export default connect(mapStoreToProps)(ExpensesSummary);


