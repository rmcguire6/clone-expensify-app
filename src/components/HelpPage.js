import React from 'react';
import { Link } from 'react-router-dom';

const HelpPage = () => (
  <div>
    Help page - <Link to="/">Go home</Link>
  </div>
);
export default HelpPage;