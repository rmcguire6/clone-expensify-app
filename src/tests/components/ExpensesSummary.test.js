import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import selectExpensesTotal from '../../selectors/expenses-total';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('ExpensesSummary should render correctly ', () => {
  const wrapper = shallow(<ExpensesSummary />);
  expect(wrapper).toMatchSnapshot();
});
test('ExpensesSummary should render correctly with one expense  ', () => {
  const expense = [{
    id: '1',
    description: 'Gum',
    note: '',
    amount: 195,
    createdAt: 0
  }]
  const wrapper = shallow(<ExpensesSummary expenses={expense} expensesTotal={1.95} />);
  expect(wrapper).toMatchSnapshot();
});
test('ExpensesSummary should render correctly with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenses={expenses} expensesTotal={94.34} />);
  expect(wrapper).toMatchSnapshot();
});

test('should output 0 if no expenses', () => {
  const total = selectExpensesTotal([]);
  expect(total).toBe(0);
})
test('should output value of the expense if expenses array has length 1', () => {
  const total = selectExpensesTotal([expenses[2]]);
  expect(total).toEqual(expenses[2].amount);
})
test('should output total if expenses has multiple entries', () => {
  const total = selectExpensesTotal(expenses);
  expect(total).toEqual(114195);
})