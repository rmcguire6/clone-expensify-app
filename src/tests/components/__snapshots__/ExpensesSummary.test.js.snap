// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpensesSummary should render correctly  1`] = `<div />`;

exports[`ExpensesSummary should render correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     expenses
     totaling   
    $94.34
  </h1>
</div>
`;

exports[`ExpensesSummary should render correctly with one expense   1`] = `
<div>
  <h1>
    Viewing 
    1
     expense
     totaling   
    $1.95
  </h1>
</div>
`;
